backends/arm/scripts/arm_model_evaluator.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import tempfile
+import zipfile
+from typing import Optional, Tuple, Union
+
+import torch
+
+
+class GenericModelEvaluator:
```

Comment:
I agree this could be moved to a more appropriate folder in `backends/arm`. I will take your suggestion of `util` .

On moving this out entirely. I would personally like to add more functionality first to see how this shapes out before moving it out. In particular, a model specific evaluator.